prism/src/main/java/org/apache/falcon/FalconServer.java

```diff
@@ -88,6 +94,14 @@ public static void main(String[] args) throws Exception {
             appPath = cmd.getOptionValue(APP_PATH);
         }
 
+        String isSafeMode = "false";
+        if (cmd.hasOption(SAFE_MODE)) {
+            isSafeMode = cmd.getOptionValue(SAFE_MODE);
+            validateSafemode(isSafeMode);
+        }
+        LOG.info("Setting system property Safemode to : {}", isSafeMode);
+        System.setProperty(StartupProperties.SAFEMODE_PROPERTY, isSafeMode);
```

Comment:
@vrangan  : As part of design doc, we said that restart will get FalconServer out of safemode. We should only consider persisting the property if we want safemode to survive restarts. If we want to make such a change in design, I request we file it as enhancement Jira. 

StartupProperties are not initialized until after falcon server is started. So if we want to use the command line args for start script to set safemode, the only way to pass that on to server is via System property. Please let me know if you have any other recommendation.